client/src/components/Form/Elements/FormCheck.vue

```diff
@@ -0,0 +1,73 @@
+<script setup>
+import { computed, ref, watch } from "vue";
+
+const emit = defineEmits(["input", "indeterminate"]);
+const props = defineProps({
+    value: {
+        default: null,
+    },
+    options: {
+        type: Array,
+        required: true,
+    },
+});
+const selectAll = ref(false);
+const indeterminate = ref(false);
+
+const currentValue = computed({
+    get: () => {
+        const val = props.value ?? [];
+        return Array.isArray(val) ? val : [val];
+    },
+    set: (val) => {
+        emit("input", val);
+    },
+});
+
+const hasOptions = computed(() => {
+    return props.options.length > 0;
+});
+
+function onSelectAll() {
+    if (selectAll.value) {
+        const allValues = props.options.map((option) => option[1]);
+        emit("input", allValues);
+    } else {
+        emit("input", []);
+    }
+}
+
+watch(currentValue, () => {
+    const valueLength = currentValue.value.length;
+    if (valueLength === 0) {
+        selectAll.value = false;
+        indeterminate.value = false;
+    } else if (valueLength === props.options.length) {
+        selectAll.value = true;
+        indeterminate.value = false;
+    } else {
+        indeterminate.value = true;
+    }
+});
+</script>
+
+<template>
+    <div v-if="hasOptions">
+        <b-form-checkbox
+            v-model="selectAll"
+            v-localize
+            size="sm"
+            class="mb-1"
+            :indeterminate="indeterminate"
+            @input="onSelectAll"
+            @change="emit('indeterminate', selectAll)">
+            Select/Unselect all
+        </b-form-checkbox>
+        <b-form-checkbox-group v-model="currentValue" stacked>
```

Comment:
```suggestion
            v-localize
            class="mb-1"
            :indeterminate="indeterminate"
            @input="onSelectAll"
            @change="emit('indeterminate', selectAll)">
            Select / Deselect all
        </b-form-checkbox>
        <b-form-checkbox-group v-model="currentValue" stacked class="pl-3">
```

The small text without spaces is a bit hard to read, and the smaller checkbox looks odd. What do you think about this design instead?

![image](https://user-images.githubusercontent.com/44241786/206714941-c157b4d1-7616-4d45-90ed-b6152fe048ba.png)

Reply:
Looks good to me, thank you for taking a look. I just changed `Deselect` to `Unselect` which seems to be more common in this case.